feature/system/control_plane_traffic/otg_tests/ingress_acl/control_plane_traffic_ingress_acl_test.go

```diff
@@ -0,0 +1,456 @@
+	term30Ipv4Ipv4 := term30Ipv4.GetOrCreateIpv4()
+	term30Ipv4Ipv4.SourceAddress = ygot.String(mgmtSrcIPv4 + "/32")
+	term30Ipv4Ipv4.Protocol = oc.UnionUint8(ipProtoICMP)
+	term30Ipv4.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_ACCEPT
+
+	// Term 40: Explicit Deny All
+	term40Ipv4 := aclSet4.GetOrCreateAclEntry(denyTermSeqID)
+	term40Ipv4.Description = ygot.String(denyTermName)
+	term40Ipv4.GetOrCreateIpv4().Protocol = oc.UnionUint8(0) // Match any protocol (0 for IPv4)
+	term40Ipv4.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_REJECT
+
+	// --- Define IPv6 ACL ---
+	aclSet6 := acl.GetOrCreateAclSet(aclNameIPv6, aclTypeIPv6)
+	aclSet6.Description = ygot.String("Control Plane Ingress IPv6 ACL")
+
+	// Term 10: Allow gRPC from Any
+	term10Ipv6 := aclSet6.GetOrCreateAclEntry(grpcTermSeqID)
+	term10Ipv6.Description = ygot.String(grpcTermName)
+	term10Ipv6.GetOrCreateTransport().DestinationPort = oc.UnionUint16(grpcPort)
+	term10Ipv6.GetOrCreateIpv6().Protocol = oc.UnionUint8(ipProtoTCP)
+	term10Ipv6.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_ACCEPT
+
+	// Term 20: Allow SSH from MGMT_SRC
+	term20Ipv6 := aclSet6.GetOrCreateAclEntry(sshTermSeqID)
+	term20Ipv6.Description = ygot.String(sshTermName)
+	term20Ipv6.GetOrCreateTransport().DestinationPort = oc.UnionUint16(sshPort)
+	term20Ipv6Ipv6 := term20Ipv6.GetOrCreateIpv6()
+	term20Ipv6Ipv6.SourceAddress = ygot.String(mgmtSrcIPv6 + "/128")
+	term20Ipv6Ipv6.Protocol = oc.UnionUint8(ipProtoTCP)
+	term20Ipv6.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_ACCEPT
+
+	// Term 30: Allow ICMPv6 from MGMT_SRC
+	term30Ipv6 := aclSet6.GetOrCreateAclEntry(icmpTermSeqID)
+	term30Ipv6.Description = ygot.String(icmpTermName)
+	term30Ipv6Ipv6 := term30Ipv6.GetOrCreateIpv6()
+	term30Ipv6Ipv6.SourceAddress = ygot.String(mgmtSrcIPv6 + "/128")
+	term30Ipv6Ipv6.Protocol = oc.UnionUint8(ipProtoICMPv6)
+	term30Ipv6.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_ACCEPT
+
+	// Term 40: Explicit Deny All
+	term40Ipv6 := aclSet6.GetOrCreateAclEntry(denyTermSeqID)
+	term40Ipv6.Description = ygot.String(denyTermName)
+	term40Ipv6.GetOrCreateIpv6().Protocol = 0 // Match any protocol (0 for IPv6 next-header)
+	term40Ipv6.GetOrCreateActions().ForwardingAction = oc.Acl_FORWARDING_ACTION_REJECT
+
+	// Push ACL configuration
+	t.Log("Pushing ACL configuration...")
+	gnmi.Replace(t, dut, gnmi.OC().Acl().Config(), acl)
+	t.Log("ACL configuration applied.")
+}
+
+// applyACLsToControlPlane applies the configured ACLs to the DUT's control plane ingress.
+func applyACLsToControlPlane(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+
+	aclSetPathIPv4 := gnmi.OC().System().ControlPlaneTraffic().Ingress().AclSet(aclNameIPv4, aclTypeIPv4)
+	// Update the 'set-name' field
+	gnmi.Update(t, dut, aclSetPathIPv4.SetName().Config(), aclNameIPv4)
+	// Update the 'type' field
+	gnmi.Update(t, dut, aclSetPathIPv4.Type().Config(), aclTypeIPv4)
+
```

Comment:
this should be a single transaction: both `set-name` and `type` are part of the aclSet key
https://github.com/openconfig/public/blob/71eae392cc60e24466fe06208dd380686524859e/release/models/system/openconfig-system-controlplane.yang#L225-L249